src/app/year-calendar/year-calendar.component.ts

```diff
@@ -0,0 +1,254 @@
+import {ECalendarValue} from '../common/types/calendar-value-enum';
+import {
+  Component,
+  EventEmitter,
+  forwardRef,
+  HostBinding,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChanges,
+  ViewEncapsulation
+} from '@angular/core';
+import {IYear} from './year.model';
+import {YearCalendarService} from './year-calendar.service';
+import * as moment from 'moment';
+import {Moment} from 'moment';
+import {IYearCalendarConfig, IYearCalendarConfigInternal} from './year-calendar-config';
+import {
+  ControlValueAccessor,
+  FormControl,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  ValidationErrors,
+  Validator
+} from '@angular/forms';
+import {CalendarValue} from '../common/types/calendar-value';
+import {UtilsService} from '../common/services/utils/utils.service';
+import {DateValidator} from '../common/types/validator.type';
+
+@Component({
+  selector: 'dp-year-calendar',
+  templateUrl: 'year-calendar.component.html',
+  styleUrls: ['year-calendar.component.less'],
+  encapsulation: ViewEncapsulation.None,
+  providers: [
+    YearCalendarService,
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => YearCalendarComponent),
+      multi: true
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => YearCalendarComponent),
+      multi: true
+    }
+  ]
+})
+export class YearCalendarComponent implements OnInit, OnChanges, ControlValueAccessor, Validator {
+  @Input() config: IYearCalendarConfig;
+  @Input() displayDate: Moment;
+  @Input() minDate: Moment;
+  @Input() maxDate: Moment;
+  @HostBinding('class') @Input() theme: string;
+
+  @Output() onSelect: EventEmitter<IYear> = new EventEmitter();
+  @Output() onNavHeaderBtnClick: EventEmitter<null> = new EventEmitter();
```

Comment:
should be void?

Reply:
Please explain

Reply:
which one more relevant `EventEmitter<null>`  or EventEmitter<void>? are you emitting null or undefined?